### DISCUSSION TITLE
Need help with custom detector

### DISCUSSION THREAD

**Discussion by ItsVaidas** [spotbugs/spotbugs#1859] [Q&A]
Hello,
I created my own custom detector and placed it into spotbugs plugins folder, but still can't find it in detectors tab.

Using:
Windows 11
Eclipse 2021-12 (4.22.0)

Detector code:

`
package my.detector;

import org.apache.bcel.classfile.JavaClass;
import org.apache.bcel.classfile.Method;

import edu.umd.cs.findbugs.BugInstance;
import edu.umd.cs.findbugs.BugReporter;
import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;

public class onEnableDisableDetector extends OpcodeStackDetector {
	
    private final BugReporter bugReporter;

    public onEnableDisableDetector(final BugReporter bugReporter) {
        this.bugReporter = bugReporter;
    }
    
    @Override
    public void visit(JavaClass obj) {
    	for (Method m : obj.getMethods())
    		if (m.getName().equals("onEnable"))
    			for (Method m2 : obj.getMethods())
    				if (m2.getName().equals("onDisable"))
    					return;
    	bugReporter.reportBug(new BugInstance(this, "ENABLE_DISABLE_BUG", LOW_PRIORITY));
    }

    @Override
    public void sawOpcode(int seen) {
    }
}
`

findbugs.xml:
`
<FindbugsPlugin xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="findbugsplugin.xsd" 
        pluginid="lt.ktu.soften.findbugs.plugingetset"
        provider="Enable Disable detector" 
        website="http://soften.ktu.lt">

        <Detector class="my.detector.onEnableDisableDetector" reports="ENABLE_DISABLE_BUG" />
        <BugPattern type="ENABLE_DISABLE_BUG" abbrev="EDB" category="CORRECTNESS"/>

</FindbugsPlugin>
`

messages.xml:
`<?xml version="1.0" encoding="UTF-8"?>
<MessageCollection xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="messagecollection.xsd">

        <Plugin>
                <ShortDescription>FindBugs get/set rule Plugin</ShortDescription>
                <Details>Provides detectors as part of the FindBugs detector</Details>
        </Plugin>

        <Detector class="my.detector.onEnableDisableDetector">
                <Details>Checks if there is onEnable, there should be onDisable</Details>

        </Detector>

        <BugPattern type="ENABLE_DISABLE_BUG">
                <ShortDescription>onEnable should have onDisable</ShortDescription>
        </BugPattern>

        <BugCode abbrev="EDB">onenable/ondisable</BugCode>
</MessageCollection>
`

![image](https://user-images.githubusercontent.com/91265795/145269148-259df20b-aed2-4cb0-b15f-93b1db04929e.png)

I'm very new to this thing. Any help would be appreciated

**Comment by S-h-i-Y-a-n-g** [comment]
Hello, I have successfully customized the rules using the official documents, and can detect errors in the idea plug-in. Your problem may be some problems with the definition of the detector
